cluster-creation-cli/README.md

```diff
@@ -0,0 +1,163 @@
+
+### Examples
+
+Ex. 1: Simple cluster creation based on a file
+
+    kubectl gs create cluster --provider capa --config myconfig.yaml
+
+Ex. 2: Simple cluster creation based on standard input
+
+    kubectl gs create cluster --provider capa <<EOF
+    global:
+      metadata:
+        servicePriority: lowest
+        name: test01
+        organization: testorg
+        description: Just a test cluster
+    EOF
+
+While ex. 1 appears simpler and allows for easier re-use of a file-based configuration for a single user, ex. 2 has the benefit of being completely self-contained. It's a pure one-liner without the need for a separate file. This way it's easy to share the command, or copy the command from a website or web UI and paste it into a terminal.
+
+Ex. 3: Overwrite a configuration value
+
+    kubectl gs create cluster --provider capa --config myconfig.yaml --set global.metadata.name=test02
+
+Ex. 4: Dry run
+
+    kubectl gs create cluster --provider capa --config myconfig.yaml --dry-run
+
+Ex. 5: Write the result to a file
+
+    kubectl gs create cluster --provider capa --config myconfig.yaml --output manifest.yaml
+
+## Configuration layering
+
+Configuration can be passed to the command in several ways:
+
+1. Any number of YAML files specified via the `--config` flag.
+2. YAML passed to standard input (STDIN).
+3. Any number of single values defined via the `--set` flag.
+
+The command will create one ConfigMap per source, and one [App](https://docs.giantswarm.io/vintage/use-the-api/management-api/crd/apps.application.giantswarm.io/) resource for the cluster-app. In the App resource, the ConfigMaps will be referenced under the `.spec.extraConfigs` property, which is an array. This way, the number of ConfigMaps is not limited.
+
+The individual ConfigMaps' priority will be configured explicitly in the following order, where later sources overwrite earlier sources:
+
+1. Each file specified via the `--config` flag, in the order they are specified.
+2. Standard input (STDIN).
+3. Each single value specified via the `--set` flag, in the order they are specified.
+
+The ConfigMap representing STDIN always comes last and takes highest priority. This allows to override configuration form one file with a another file specified later in the chain, or with a value passed via STDIN.
+
+In other words, we leverage the App platform's configuration layering principle to provide a flexible templating mechanism, where the underlying layer is always provided by the app's default values.
+As a usage example, this layering could be used to create a cluster using a template configuration file, and overriding some specific values via STDIN:
+
+    kubectl gs create cluster --provider capa --config template.yaml <<EOF
+    global:
+      metadata:
+        name: test02
+    EOF
+
+Alternative syntax for above example, using the `--set` flag:
```

Comment:
Does each `--set` property generate a new ConfigMap or do that all get merged into one?

Reply:
IMO everything coming from `--set` should end up in one common ConfigMap.

Reply:
Clarified in 9911b567a65f912e1aeb7bed2330fca0380dd07c